src/libraries/System.Runtime.InteropServices/tests/ComInterfaceGenerator.Tests/RcwAroundCcwTests.cs

```diff
@@ -175,7 +176,8 @@ public void IStringArrayMarshallingFails()
             {
                 obj.InParam(in strings);
             });
-            Assert.Throws<MarshallingFailureException>(() =>
+            // ThrowForHR will throw an untyped Exception
```

Comment:
This is an implementation detail of this test?

Reply:
Since this doesn't marshal in, this doesn't fail until the UnmanagedToManaged stub tries to marshal it back, so the COM method returns an HResult and throws for the HR. I assumed from the test failure it would be an untyped exception, but looks like it should be a ComException.

I'll add more details in the comment and update to ComException.

Reply:
Ah, it was non-windows that was failing. On windows it properly unwraps the exception, but on osx and linux it creates a plain `System.Exception`. I broke it out into two tests with `ConditionalFact`s